gym/lib/gym/runner.rb

```diff
@@ -168,6 +168,18 @@ def copy_mac_app
       app_path
     end
 
+    # Finds the most likely candidate for this target's .app
+    def find_app_path
+      # guess the correct .app name from the config
+      output_path = File.join(BuildCommandGenerator.archive_path, "Products/Applications", Gym.config[:output_name] + ".app")
+      app_path = output_path if File.exist?(output_path)
```

Comment:
We should show a `UI.user_error!` here if the `Gym.config[:output_name]` value is given but no file is found. Consider the following example:

```bash
$ gym --output_name "TetsApp.app" --scheme "TestApp" #...snip
# Produces TestApp.app and HelperForTestApp.app, the command will seemingly work correctly 
$ gym  --output_name "HelperForTetsApp.app" --scheme "TestApp" #...snip
# Produces TestApp.app and HelperForTestApp.app, the command will copy the wrong .app ... User starts pulling their hair out because they can't figure out why it works sometimes and not other times
```